disable modify yaml constructor

clan opened the pull request:
which will modify the default behavior of yaml load.

Foe example, for following example (t.sls), it will cause the difference
between the content of file testa and testb, but it should be identical!

$ cat t

> {%- load_yaml as vars %}
> toaddr:
>   \- test@test.com
> {%- endload -%}
> {{ vars.toaddr }}

$ cat t.sls

> /tmp/testa:
>   file.managed:
>     - source: salt://t
>     - user: root
>     - group: root
>     - mode: "0755"
>     - template: jinja
> 
> sys-power/acpid:
>   pkg.installed:
>     - refresh: False
> 
> /tmp/testb:
>   file.managed:
>     - source: salt://t
>     - user: root
>     - group: root
>     - mode: "0755"
>     - template: jinja

$ touch /tmp/test{a,b}
$ salt-call state.sls t

> local:
> ----------
>           ID: /tmp/testa
>     Function: file.managed
>       Result: None
>      Comment: The file /tmp/testa is set to be changed
>      Changes:
>               ----------
>               diff:
>                   ---
>                   +++
>                   @@ -0,0 +1 @@
>                   +['test@test.com']
> ----------
>           ID: /tmp/testb
>     Function: file.managed
>       Result: None
>      Comment: The file /tmp/testb is set to be changed
>      Changes:
>               ----------
>               diff:
>                   ---
>                   +++
>                   @@ -0,0 +1 @@
>                   +[u'test@test.com']

salt-jenkins commented:
@clan, thanks for your PR! By analyzing the history of the files in this pull request, we identified @homolkad, @terminalmage and @rallytime to be potential reviewers.

clan commented:
We don't use heat state/module, so I don't know will this break any function.
Maybe the commiter of heat module can provide better solution. Anyway, it's best to do not change the default behavior of a widely used module (yaml in this case).

cachedout commented:
@homolkad Can you comment on this at all?

homolkad commented:
Yaml constructor change has been used as "Language-Independen string", but I think it would be better this change are removed.
@clan has right, it's better to do not changed the default behavior of a widely used module. 
We will rewritted our template.yaml for this chtange.

clan commented:
@homolkad cool, so what should we do next for this pr?

homolkad commented:
@clan for me is this pr correct and acceptable

clan commented:
@homolkad maybe you need to approve this pr before it can be merged, thanks.

cachedout commented on salt/modules/heat.py:
```diff
@@ -102,8 +102,8 @@ def _construct_yaml_str(self, node):
     return self.construct_scalar(node)
-YamlLoader.add_constructor(u'tag:yaml.org,2002:str',
-                           _construct_yaml_str)
+# YamlLoader.add_constructor(u'tag:yaml.org,2002:str',
```
We don't like to have dead code in our code base. Instead of commenting these out, please just remove these lines.

cachedout commented on salt/states/heat.py:
```diff
@@ -80,8 +80,8 @@ def _construct_yaml_str(self, node):
     return self.construct_scalar(node)
-YamlLoader.add_constructor(u'tag:yaml.org,2002:str',
-                           _construct_yaml_str)
+# YamlLoader.add_constructor(u'tag:yaml.org,2002:str',
```
Same here. Please just remove these lines. Thanks.

clan committed:
remove modify yaml constructor

which will modify the default behavior of yaml load.

Foe example, for following example (t.sls), it will cause the difference
between the content of file testa and testb, but it should be identical!

$ cat t
{%- load_yaml as vars %}
toaddr:
  - test@test.com
{%- endload -%}
{{ vars.toaddr }}
$ cat t.sls
/tmp/testa:
  file.managed:
    - source: salt://t
    - user: root
    - group: root
    - mode: "0755"
    - template: jinja

sys-power/acpid:
  pkg.installed:
    - refresh: False

/tmp/testb:
  file.managed:
    - source: salt://t
    - user: root
    - group: root
    - mode: "0755"
    - template: jinja
$ touch /tmp/test{a,b}
$ salt-call state.sls t
local:
----------
          ID: /tmp/testa
    Function: file.managed
      Result: None
     Comment: The file /tmp/testa is set to be changed
     Changes:
              ----------
              diff:
                  ---
                  +++
                  @@ -0,0 +1 @@
                  +['test@test.com']
----------
          ID: /tmp/testb
    Function: file.managed
      Result: None
     Comment: The file /tmp/testb is set to be changed
     Changes:
              ----------
              diff:
                  ---
                  +++
                  @@ -0,0 +1 @@
                  +[u'test@test.com']

Commit:
Merge branch '2017.7' into yaml

clan commented:
updated.

In my opinion, my pr only fix the bug shown in the commit msg. Because I don't have test case for this module, so it's better to remove or rewrite all the code which modify the default yaml behavior by @homolkad .
